Wikipedia/Code/FindInPageKeyboardBar.h

```diff
@@ -0,0 +1,24 @@
+#import <UIKit/UIKit.h>
+
+@class FindInPageKeyboardBar;
+
+@protocol FindInPageBarDelegate <NSObject>
+
+- (void)findInPageTermChanged:(NSString *)text sender:(FindInPageKeyboardBar *)sender;
+- (void)findInPageCloseButtonTapped;
+- (void)findInPageClearButtonTapped;
+- (void)findInPagePreviousButtonTapped;
+- (void)findInPageNextButtonTapped;
+
+@end
+
+@interface FindInPageKeyboardBar : UIInputView
+
+@property (weak, nonatomic) id<FindInPageBarDelegate> delegate;
+
+@property (strong, nonatomic) IBOutlet UITextField *textField;
```

Comment:
Its nice to hide this from outside observers… this encourages outside users to "touch it".

If other objects need something from it, its nice to provide an API for them to get/do what they need so that this class can be wholly responsible for managing the text field and no other objects can do anything unexpected with it.